src/metrics.go

```diff
@@ -16,18 +16,27 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+type metricsCollector struct {
+	integration         *integration.Integration
+	db                  database.DBWrapper
+	wg                  *sync.WaitGroup
+	instanceLookUp      map[string]string
+	customMetricsQuery  string
+	customMetricsConfig string
+}
+
 // collectMetrics spins off goroutines for each of the metric groups, which
 // send their metrics to the populateMetrics goroutine
-func collectMetrics(db database.DBWrapper, populaterWg *sync.WaitGroup, i *integration.Integration, instanceLookUp map[string]string, customMetricsQuery string, customMetricsConfig string) {
-	defer populaterWg.Done()
+func (mc *metricsCollector) collectMetrics() {
+	defer mc.wg.Done()
```

Comment:
Nit: Since the struct is already called metricsCollecor, I would just call the method collect

Reply:
that make totally sense :) I will change it thanks